mitmproxy/contentviews/base.py

```diff
@@ -0,0 +1,65 @@
+# Default view cutoff *in lines*
+
+from typing import Iterable, AnyStr, List
+from typing import Mapping
+from typing import Tuple
+
+VIEW_CUTOFF = 512
+
+KEY_MAX = 30
+
+
+class View:
```

Comment:
Presumably not a refactoring you're going to be doing in this patch, but we need a clear interface for Views. I suggest that each view exposes two class methods: 

- ```check(f) -> Optional[View]```, which takes an object (flow, request, response), does whatever verification is needed to see whether it can sensibly render it, and returns an instantiated class.
- ```force(f) -> Optional[View]```, which tries to force the view to render the object, regardless of whether the sanity-check succeeded. This is still optional, because for some things the answer will still be "no". 

This means that each view can now have a natural set of instantiation parameters, rather than the nebulous and poorly defined "metadata" cludge. We can also use the check() method in places like mitmproxy console to only display a list of views that "make sense", cutting down on the very long list of elements we currently have. 

This gets rid of all the arcane mechanisms we currently have like AutoView and centralised content guessing, instead letting each View define their own semantics. When automatically choosing a View, the ViewManager can use the first matching view, and we can use view order to control the priorities.